library/java/src/io/envoyproxy/envoymobile/engine/JniLibrary.java

```diff
@@ -3,25 +3,24 @@
 import io.envoyproxy.envoymobile.engine.types.EnvoyData;
 import io.envoyproxy.envoymobile.engine.types.EnvoyHeaders;
 import io.envoyproxy.envoymobile.engine.types.EnvoyObserver;
-import io.envoyproxy.envoymobile.engine.types.EnvoyStream;
 
 class JniLibrary {
 
-  protected static native EnvoyStream startStream(EnvoyObserver observer);
+  protected static native long startStream(EnvoyObserver observer);
 
-  protected static native int sendHeaders(EnvoyStream stream, EnvoyHeaders headers,
-                                          boolean endStream);
+  protected static native int sendHeaders(long stream, EnvoyHeaders headers, boolean endStream);
 
-  protected static native int sendData(EnvoyStream stream, EnvoyData data, boolean endStream);
+  protected static native int sendData(long stream, EnvoyData data, boolean endStream);
 
-  protected static native int sendMetadata(EnvoyStream stream, EnvoyHeaders metadata,
-                                           boolean endStream);
+  protected static native int sendMetadata(long stream, EnvoyHeaders metadata);
 
-  protected static native int sendTrailers(EnvoyStream stream, EnvoyHeaders trailers);
+  protected static native int sendTrailers(long stream, EnvoyHeaders trailers);
 
-  protected static native int locallyCloseStream(EnvoyStream stream);
+  protected static native int locallyCloseStream(long stream);
 
-  protected static native int resetStream(EnvoyStream stream);
+  protected static native int resetStream(long stream);
 
   protected static native int runEngine(String config, String logLevel);
```

Comment:
Maybe move this out from the middle of the stream interface and delineate with comments.

Reply:
mirroring the `main_interface.h`